privacyidea/lib/smsprovider/SMSProvider.py

```diff
@@ -170,6 +179,7 @@ def set_smsgateway(identifier, providermodule, description=None,
     smsgateway = SMSGateway(identifier, providermodule,
```

Comment:
``SMSGateway.__init__`` already calls ``.save()`` to write the entry to the database, even if ``check_configuration`` fails later. This means that even though an error is raised, we will end up with invalid database entries. Should we change that, or just keep this behavior?

Reply:
We need to save this, otherwise we can not check the config.
Before the configuration was always saved without checking it.
Yes, we need to keep it that way.

Reply:
My issue with this is as follows: I enter an invalid firebase configuration. The check fails, which causes the HTTP request to fail with a 500 response. Because of this response, I would expect that the failed request does not actually save the invalid configuration.

I think checking before saving would be nice. But it seems it would require moving around a lot of code, so let's keep it like above for now.